internal/replication/blockreplicator.go

```diff
@@ -35,6 +36,12 @@ type BlockLedgerReader interface {
 	Get(blockNumber uint64) (*types.Block, error)
 }
 
+//go:generate counterfeiter -o mocks/pending_txs.go --fake-name PendingTxsReleaser . PendingTxsReleaser
+
+type PendingTxsReleaser interface {
+	ReleaseWithError(txIDs []string, err error)
+}
```

Comment:
There are two options. Either use the real thing, queue.PendingTxs, which is a bit cumbersome, or use a mock. I prefer a mock. When it comes to mocks, I always prefer an automatically generated one. I think that interface is a small price to pay, and is actually the go idiomatic way for testing, AFAIK. So I prefer to leave it as is, since simplicity is key in this package, as I am going to add a lot more tests as we role out new capabilities.